internal/drivers/manager.go

```diff
@@ -356,6 +357,13 @@ func (m *Manager) Destroy(ctx context.Context, poolName, instanceID string) erro
 		return err
 	}
 
+	stageRuntimeID := instance.Stage
+	cleanupVm := debugState.GetDebugState().Get(stageRuntimeID)
```

Comment:
check if entry is present or use field name as retainVm so that its default value means destroying the vm

Reply:
value is coming as true if the entry is not present.

Reply:
Better to use retain vm instead since we mostly assume that default value is false.

Reply:
makes sense, making the change